mysql/datadog_checks/mysql/statement_samples.py

```diff
@@ -667,11 +696,14 @@ def _explain_statement(self, cursor, statement, schema, obfuscated_statement):
 
         if not self._can_explain(obfuscated_statement):
             self._log.debug('Skipping statement which cannot be explained: %s', obfuscated_statement)
-            return None
+            return None, DBExplainError.no_plans_possible, None
 
-        if self._collection_strategy_cache.get(strategy_cache_key) == explain_strategy_error:
-            self._log.debug('Skipping statement due to cached collection failure: %s', obfuscated_statement)
-            return None
+        strategy_cache = self._collection_strategy_cache.get(strategy_cache_key)
+        if strategy_cache:
+            strategy_error, db_explain_error, err = strategy_cache
+            if strategy_error == explain_strategy_error:
```

Comment:
Do we still need `explain_strategy_error` or is it enough to store `DBExplainError` in there?